robot/web/stream/server.go

```diff
@@ -412,3 +486,97 @@ func (server *Server) removeMissingStreams() {
+func (server *Server) refreshAudioSources() {
+	for _, name := range audioinput.NamesFromRobot(server.robot) {
+		input, err := audioinput.FromRobot(server.robot, name)
+		if err != nil {
+			continue
+		}
+		existing, ok := server.audioSources[input.Name().SDPTrackName()]
+		if ok {
+			existing.Swap(input)
+			continue
+		}
+		newSwapper := gostream.NewHotSwappableAudioSource(input)
+		server.audioSources[input.Name().SDPTrackName()] = newSwapper
+	}
+}
+
+func (server *Server) createStream(config gostream.StreamConfig, name string) (gostream.Stream, bool, error) {
+	stream, err := server.NewStream(config)
+	// Skip if stream is already registered, otherwise raise any other errors
+	var registeredError *StreamAlreadyRegisteredError
+	if errors.As(err, &registeredError) {
+		server.logger.Debugw("stream already registered", "name", name)
+		return nil, true, nil
+	} else if err != nil {
+		return nil, false, err
+	}
+	return stream, false, err
+}
+
+func (server *Server) startStream(streamFunc func(opts *BackoffTuningOptions) error) {
+	waitCh := make(chan struct{})
+	server.activeBackgroundWorkers.Add(1)
+	utils.PanicCapturingGo(func() {
+		defer server.activeBackgroundWorkers.Done()
+		close(waitCh)
+		if err := streamFunc(&BackoffTuningOptions{}); err != nil {
+			if utils.FilterOutError(err, context.Canceled) != nil {
+				server.logger.Errorw("error streaming", "error", err)
+			}
+		}
+	})
+	<-waitCh
+}
+
+func (server *Server) startVideoStream(ctx context.Context, source gostream.VideoSource, stream gostream.Stream) {
+	server.startStream(func(opts *BackoffTuningOptions) error {
+		streamVideoCtx, _ := utils.MergeContext(server.closedCtx, ctx)
+		// Use H264 for cameras that support it; but do not override upstream values.
+		if props, err := server.propertiesFromStream(ctx, stream); err == nil && slices.Contains(props.MimeTypes, rutils.MimeTypeH264) {
+			streamVideoCtx = gostream.WithMIMETypeHint(streamVideoCtx, rutils.WithLazyMIMEType(rutils.MimeTypeH264))
+		}
+		return streamVideoSource(streamVideoCtx, source, stream, opts, server.logger)
+	})
+}
+
+func (server *Server) startAudioStream(ctx context.Context, source gostream.AudioSource, stream gostream.Stream) {
+	server.startStream(func(opts *BackoffTuningOptions) error {
+		// Merge ctx that may be coming from a Reconfigure.
+		streamAudioCtx, _ := utils.MergeContext(server.closedCtx, ctx)
+		return streamAudioSource(streamAudioCtx, source, stream, opts, server.logger)
+	})
+}
+
+func (server *Server) propertiesFromStream(ctx context.Context, stream gostream.Stream) (camera.Properties, error) {
+	res, err := server.robot.ResourceByName(camera.Named(stream.Name()))
+	if err != nil {
+		return camera.Properties{}, err
+	}
+
+	cam, ok := res.(camera.Camera)
+	if !ok {
+		return camera.Properties{}, errors.Errorf("cannot convert resource (type %T) to type (%T)", res, camera.Camera(nil))
+	}
+	return cam.Properties(ctx)
```

Comment:
huh.... do we care about any properties other than frame rate for the streams?

Reply:
Yeah we should only care about frame rate. I am tempted to take this properties call out for now since this code path for h264 mimetype handling is not being used.

Reply:
let's just take it out then.

Reply:
Ok, [took out](https://github.com/viamrobotics/rdk/pull/4490/commits/1a6a1fd0b821caae9a8702785bd673841ded08b7) the h264 mimetype handler and `propertiesFromStream` helper.